internal/core/processor/evolution/client_wb_test.go

```diff
@@ -0,0 +1,44 @@
+package evolution
+
+import (
+	"math/big"
+	"testing"
+)
+
+func TestHexToDecimal(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		input     string
+		expected  *big.Int
+		expectErr bool
+	}{
+		{"1a", big.NewInt(26), false},
+		{"0x1a", big.NewInt(26), false},
+		{"abc", big.NewInt(2748), false},
+		{"0xabc", big.NewInt(2748), false},
+		{"invalid", nil, true},   // Invalid hex
```

Comment:
If the `hexToDecimal` functionis unable to convert to a big.Int, it will return both a nil value and an error. Could you please specify the exact aspect that is raising concerns for you?